Upgrade setuptools.

jdm opened the pull request:
Attempting to fix a new build error in CI.

jdm committed:
Install setuptools.

jdm committed:
Upgrade setuptools.

jdm committed:
Remove explicit setuptools installation.

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 5bebdea has been approved by `jdm`

<!-- @bors-servo r=jdm 5bebdeadbe2f9ceb2a0b828ccac4068934d62124 -->

bors-servo commented:
:hourglass: Testing commit 5bebdeadbe2f9ceb2a0b828ccac4068934d62124 with merge 9c8493be4ae8ea8e75aef5d01ce60aa67667428f...

bors-servo commented:
:broken_heart: Test failed - [checks-github](https://github.com/servo/servo/runs/8179599877?check_suite_focus=true)

jdm commented:
@bors-servo retry

bors-servo commented:
:hourglass: Testing commit 5bebdeadbe2f9ceb2a0b828ccac4068934d62124 with merge a8db50e9f47c8ffb2bc30a7b2c9595da818d4e44...

jdm commented:
```
  Creating Python 3 environment
  created virtual environment CPython3.10.6.final.0-64 in 2352ms
    creator CPython3Posix(dest=/Users/runner/work/servo/servo/target/release/build/mozjs_sys-10fa456ddf80aa49/out/build/_virtualenvs/init_py3, clear=False, no_vcs_ignore=False, global=False)
    seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/runner/Library/Application Support/virtualenv)
      added seed packages: pip==20.3.1, setuptools==51.0.0, wheel==0.36.1
    activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
  DEBUG: python3: running with pid 40742
  DEBUG: python3: sys.executable: '/Users/runner/work/servo/servo/python/_virtualenv3.10/bin/python3'
  DEBUG: python3: executable from configuration: None
  DEBUG: python3: found executable: '/Users/runner/work/servo/servo/python/_virtualenv3.10/bin/python3'
  INFO: Creating Python 3 environment
  INFO: created virtual environment CPython3.10.6.final.0-64 in 2352ms
  INFO:   creator CPython3Posix(dest=/Users/runner/work/servo/servo/target/release/build/mozjs_sys-10fa456ddf80aa49/out/build/_virtualenvs/init_py3, clear=False, no_vcs_ignore=False, global=False)
  INFO:   seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/runner/Library/Application Support/virtualenv)
  INFO:     added seed packages: pip==20.3.1, setuptools==51.0.0, wheel==0.36.1
  INFO:   activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

  --- stderr
  Traceback (most recent call last):
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/js/src/../../configure.py", line 231, in <module>
      sys.exit(main(sys.argv))
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/js/src/../../configure.py", line 55, in main
      sandbox.run(os.path.join(os.path.dirname(__file__), "moz.configure"))
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 502, in run
      self.include_file(path)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 493, in include_file
      exec_(code, self)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/util.py", line 55, in exec_
      exec(object, globals, locals)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/moz.configure", line 7, in <module>
      include("build/moz.configure/init.configure")
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 853, in include_impl
      self.include_file(what)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 493, in include_file
      exec_(code, self)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/util.py", line 55, in exec_
      exec(object, globals, locals)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/build/moz.configure/init.configure", line 241, in <module>
      def virtualenv_python3(env_python, virtualenv_name, build_env, mozconfig, help):
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 836, in decorator
      depends = DependsFunction(self, func, dependencies, when=when)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 137, in __init__
      sandbox._value_for(self)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 614, in _value_for
      return self._value_for_depends(obj)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/util.py", line 1049, in method_call
      cache[args] = self.func(instance, *args)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 623, in _value_for_depends
      value = obj.result()
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/util.py", line 1049, in method_call
      cache[args] = self.func(instance, *args)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 162, in result
      return self._func(*resolved_args)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/configure/__init__.py", line 1298, in wrapped
      return new_func(*args, **kwargs)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/build/moz.configure/init.configure", line 354, in virtualenv_python3
      manager.build(python)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/virtualenv.py", line 551, in build
      result = self._log_process_output(args, cwd=self.topsrcdir)
    File "/Users/runner/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/4cd3f5b/mozjs/python/mozbuild/mozbuild/virtualenv.py", line 244, in _log_process_output
      proc = subprocess.Popen(
    File "/usr/local/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 969, in __init__
      self._execute_child(args, executable, preexec_fn, close_fds,
    File "/usr/local/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 1845, in _execute_child
      raise child_exception_type(errno_num, err_msg, err_filename)
  FileNotFoundError: [Errno 2] No such file or directory: '/Users/runner/work/servo/servo/target/release/build/mozjs_sys-10fa456ddf80aa49/out/build/_virtualenvs/init_py3/bin/python'
  make: *** [maybe-configure] Error 1
```

bors-servo commented:
:broken_heart: Test failed - [checks-github](https://github.com/servo/servo/runs/8182425097?check_suite_focus=true)

yvt commented on python/requirements.txt:
```diff
@@ -6,7 +6,7 @@ distro == 1.4
 mozinfo == 1.2.1
 mozlog == 7.1.0
-setuptools == 60.8.1
+setuptools == 65.3.0
```
```suggestion
```

If I understand correctly, specifying setuptools in `python/requirements.txt` is not needed at all because virtualenv installs a bundled version of setuptools by default. (Assuming this file is solely used to set up a virtualenv environment.)

It may be possible that specifying it here results in a corrupted installation of setuptools. When I reproduced this issue, `servo/python/_virtualenv3.10/lib/python3.10/site-packages/setuptools` contained files from different versions of setuptools ([`_path.py`](https://github.com/pypa/setuptools/blob/v60.8.1/setuptools/_path.py) from v60.8.1 and [`config/expand.py`](https://github.com/pypa/setuptools/blob/v65.3.0/setuptools/config/expand.py#L48) from v65.3.0).

yvt committed:
chore(ci): relink `python3` to `python3.9` on Homebrew

Implements a work-around for the compatibility issue between mozjs and
Homebrew's Python 3.10 formula.

yvt commented:
I think downgrading (pinning) to Python 3.9 is a reasonable short-term approach to temporarily work around the mozjs issue and get CI running again for now. Could you add <https://github.com/yvt/servo/commit/12bb89af108cbed2095318d60ddb4e984e2aa0c3> to the PR? I'm sure bors won't let us merge the fixes for these two issues separately.

bors-servo commented:
:hourglass: Testing commit f47a53f3b1a6f41711d02d59ee4c7de27c215d98 with merge c1edf6977844f0d863d8192503fc7b77dbed3597...

bors-servo commented:
:sunny: Test successful - [checks-github](https://github.com/servo/servo/actions/runs/3116571763/jobs/5055063174)
Approved by: jdm
Pushing c1edf6977844f0d863d8192503fc7b77dbed3597 to master...
